format/Flight.proto

```diff
@@ -301,6 +301,33 @@ message Location {
  */
 message Ticket {
   bytes ticket = 1;
+  // Data representing some part of the data retrievable by the ticket.
+  //
+  // `inlined_completeness` indicates what part of the data retriavable
+  // by the ticket this represents. This is provided as an optimization for
+  // client/server applications that want to reduce latency to first result
+  // without requiring another RPC round-trip to retrieve the ticket.  applications
+  // built on top of Flight are responsible for any negotiation necessary on whether
+  // inlining data is appropriate.
+  //
+  // The size of inlined_data is expected to be small (typically less then 1MB) 
+  // and inlining too much data across tickets can run into underlying transport 
+  // limitations.  Furthermore, since the data is expected to be small, implementations 
+  // are less likely to optimize for zero-copy in these cases.  
+  repeated FlightData inlined_data = 2;
+  enum InlinedCompleteness {
+    // Default is no data is inlined.  An UNDEFINED value is not provided because an enum
+    // for the data that the client isn't aware of makes the data unusable.
+    NO_INLINED_DATA = 0;
+    // The data present in inlined_data represents all data
+    // present for the ticket.
+    COMPLETE_DATA = 1;
+    // The data present for inlined_data represents only a partial
+    // sample of the data available for the ticket.  No guarantees on the 
+    // ordering are provided.
+    SAMPLE_DATA = 2;
```

Comment:
Ah, so the semantics for this would be perhaps a `.head()` or `LIMIT <n>` so a client can preview some arbitrary amount of data and call DoGet for the entire dataset?

Reply:
Yeah i think this is what @BryanCutler  suggested on the mailing list thread.

Reply:
What would be a usecase for `SAMPLE_DATA`?

I'm thinking that something like `LIMIT` would probably come as part of the query, so for IE `LIMIT 10` it would be `COMPLETE_DATA` but the limit is implemented data-side

Reply:
the use-case on the mailing list was data preview.  @BryanCutler might have something more specific.  But one can imagine wanting to show a few rows of data in a UI for users with lowest possible latency while all data is fetched.

Reply:
I feel like this type of behavior could also be encoded using application specific metadata (rather than explicitly in the Flight protocol itself)

Reply:
FlightInfo lacks application specific metadata, though.